src/components/molecules/EarthEditor/PropertyPane/PropertyField/ColorField/hooks.ts

```diff
@@ -0,0 +1,150 @@
+import { useState, useEffect, useCallback, useRef } from "react";
+import tinycolor, { ColorInput } from "tinycolor2";
+
+export type RGBA = {
+  r: number;
+  g: number;
+  b: number;
+  a: number;
+};
+
+export type Params = {
+  value?: string;
+  onChange?: (value: string | null) => void | undefined;
+};
+
+export default ({ value, onChange }: Params) => {
+  const [colorState, setColor] = useState<string | null>(null);
+  const [rgba, setRgba] = useState<RGBA>(tinycolor(value).toRgb());
+  const [open, setOpen] = useState(false);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const pickerRef = useRef<HTMLDivElement>(null);
+
+  const getHexString = (value?: ColorInput) => {
+    if (!value) return undefined;
+    const color = tinycolor(value);
+    return color.getAlpha() === 1 ? color.toHexString() : color.toHex8String();
+  };
+
+  const handleChange = useCallback(newColor => {
+    const color = getHexString(newColor);
+    if (!color) return;
+    setColor(color);
+    setRgba(newColor);
+  }, []);
```

Comment:
Is this OK though it  depends on `setColor`, `setRgba`?

Reply:
As far as I understand it setColor and setRgba are the setter functions and will never change (they aren't props so no chance of them not being available either) so they aren't considered dependencies. Dependencies for a useCallback I believe will only be variables or functions that have a chance of changing.

Reply:
Oh, yes that's true. Sorry my bad!